src/visual/dataColor.js

```diff
@@ -74,6 +77,25 @@ export default function (seriesType) {
                     // Set data all color for legend
                     dataAll.setItemVisual(rawIdx, 'color', singleDataColor);
                 }
+
+                if (!singleDataBorderColor) {
+                    var borderColor = itemModel.get('itemStyle.borderColor')
+                    || seriesModel.getColorFromPalette(
+                        dataAll.getName(rawIdx) || (rawIdx + ''), seriesModel.__paletteScope,
+                        dataAll.count()
```

Comment:
`seriesModel.getColorFromPalette` should better not repeat,
now that it both used in `if (!singleDataColor)` and `if (!singleDataBorderColor)`.